ts/test/session/sending/PendingMessageCache_test.ts

```diff
@@ -0,0 +1,203 @@
+import { expect } from 'chai';
+
+import * as MessageUtils from '../../../session/utils';
+import { TestUtils } from '../../../test/test-utils';
+import { PendingMessageCache } from '../../../session/sending/PendingMessageCache';
+
+// Equivalent to Data.StorageItem
+interface StorageItem {
+  id: string;
+  value: any;
+}
+
+describe('PendingMessageCache', () => {
+  // Initialize new stubbed cache
+  let pendingMessageCacheStub: PendingMessageCache;
+
+  beforeEach(async () => {
+    // Stub out methods which touch the database
+    const storageID = 'pendingMessages';
+    let data: StorageItem = {
+      id: storageID,
+      value: '',
+    };
+
+    TestUtils.stubData('getItemById')
+      .withArgs('pendingMessages')
+      .callsFake(async () => {
+        return data;
+      });
+
+    TestUtils.stubData('createOrUpdateItem').callsFake((item: StorageItem) => {
+      if (item.id === storageID) {
+        data = item;
+      }
+    });
+
+    pendingMessageCacheStub = new PendingMessageCache();
+    await pendingMessageCacheStub.isReady;
+  });
+
+  afterEach(() => {
+    TestUtils.restoreStubs();
+  });
+
+  it('can initialize cache', async () => {
```

Comment:
Maybe add a small test for cache initialization from db?

So if you have some data in db, check if initialising restores back the pending messages.